### PR TITLE
Cloudbuilds

### PR DESCRIPTION
**Pull request by jurgelenas**
# General

- [x] BinaryConfigurator `IsCompatible` cloudbuilds detection
- [x] Merge 2.4 GHz and 900 MHz categories in `devices/` 
- [x] Implement cloud cache downloads from Github releases
- [x] Detect `FirmwareSource.Local` directories that are compatible with cached binaries (for users behind the restrictive firewalls)
- [x] Implement checks for features / modes that are incompatible with cloud builds cache (airport, txt user defines)
- [x] Plan for future backpack support
- [x] Use artifactory.expresslrs.org domain to download firmware artifacts

# Linux

- [x] Implement `pip install pyserial esptool` (make sure to use pinned versions) or add these requirements to package documentation. This could get complicated if we need to create `venv` for `expresslrs-configurator` like `platformio` does.

# Windows

- [x] Update portable python to include pyserial and esptool pip packages (https://github.com/ExpressLRS/ExpressLRS-Configurator/blob/master/dependencies/windows_amd64/portablepy.ps1)

# macOS

- [ ] Test that unified cloud build works properly

### CODE REVIEW

File: src/api/src/services/BinaryFlashingStrategy/BinaryConfigurator/index.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      flags: maskSensitiveFlags(flags),
+      hardwareDefinitionsPath,
+      firmwareBinaryPath,
+    });
+    const binaryConfiguratorArgs = [
+      ...this.stringifyFlags([...flags, ['--dir', hardwareDefinitionsPath]]),
+      firmwareBinaryPath,
+    ];
+    return this.python.runPythonScript(
+      path.join(firmwareSourcePath, 'python', 'binary_configurator.py'),
```
Comment (pkendall64) [inline src/api/src/services/BinaryFlashingStrategy/BinaryConfigurator/index.ts:167]: This should now be using the flasher.pyz file included in the firmware download zip, all the parameters are the same. flasher.pyz is a packages version of the python folder from the main firmware repo will all it's dependencies included.  Check the PR https://github.com/ExpressLRS/ExpressLRS/pull/2145 (also you should be able to test using the zip from the build for that PR too).

**Comment by razzeee** [comment]
Could you give me some hints, where to find this and test it?

**Comment by jurgelenas** [comment]
> Could you give me some hints, where to find this and test it?

https://github.com/ExpressLRS/ExpressLRS-Configurator/releases/tag/v1.6.0-rc.1 :)

It is so much better.

**Comment by razzeee** [comment]
Yes, I've build that as a flatpak for flathub beta, but I'm failing to see any difference in the UI or would it still say `Build and Flash` and but don't build?

**Comment by jurgelenas** [comment]
> Yes, I've build that as a flatpak for flathub beta, but I'm failing to see any difference in the UI or would it still say `Build and Flash` and but don't build?

Everything is behind the scenes. There are almost no visual changes.

But you should notice that the build speed is greatly improved. Firmware source is no longer compiled on the spot.

**Comment by razzeee** [comment]
Okay, I'll just try to flash something then. Would have expected the buttons to not say `Build` anymore :sweat_smile:

**Pull request merged**